### ISSUE TITLE
[Accessibility]: Garbled link description for code intel detail views

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35183]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518

### Problem description

The link to "detail" view of code intel objects are a huge mouthful for the screen reader. They read out the relative link, which contains the GraphQL ID.

<img width="1164" alt="Screen Shot 2022-05-09 at 4 27 28 PM" src="https://user-images.githubusercontent.com/103087/167501729-923e61d3-b83a-4515-8f70-c62896e5d98c.png">

This applies to:

- Code intelligence configuration policies (site-admin and repository views)
- Code intelligence uploads (site-admin and repository views)
- Code intelligence index job records (site-admin and repository views)

### Expected behavior

The link should be read as something more human distinguishable.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by macraig** [comment]
Closing as identified issues have been completed

**Comment by sourcegraph-bot-2** [comment]
Heads up @taylorsperry @jasongornall - the "team/frontend-platform" label was applied to this issue.
